Packs/BitwardenPasswordManager/Integrations/BitwardenPasswordManagerEventCollector/BitwardenPasswordManagerEventCollector.py

```diff
@@ -0,0 +1,280 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from CommonServerUserPython import *  # noqa
+
+VENDOR = 'Bitwarden'
+PRODUCT = 'Password Manager'
+
+DEFAULT_MAX_FETCH = 500
+DEFAULT_FIRST_FETCH = '3 days'
+MINUTES_BEFORE_TOKEN_EXPIRED = 2
+
+AUTHENTICATION_FULL_URL = 'https://identity.bitwarden.com/connect/token'
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+class Client(BaseClient):
+    """
+    Client class to interact with the service API
+    """
+
+    def __init__(self, base_url: str, verify: bool, proxy: bool, client_id: str, client_secret: str):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.token = self.login(client_id, client_secret)
+
+    def login(self, client_id: str, client_secret: str) -> str:
+        integration_context = get_integration_context()
+        demisto.log(f"{integration_context=}")
+        if token := integration_context.get('token'):
+            expires_date = integration_context.get('expires')
+            if expires_date and not self.is_token_expired(expires_date):
+                demisto.log(f"Token is valid: {expires_date=}, {token=}")
+                return token
+
+        json_data = {
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'grant_type': 'client_credentials',
+            'scope': 'api.organization'
+        }
+        return self.create_new_token(json_data)
+
+    def is_token_expired(self, expires_date: str) -> bool:
+        utc_now = get_current_time()
+        expires_datetime = arg_to_datetime(expires_date)
+        return utc_now > expires_datetime
+
+    def create_new_token(self, json_data: dict) -> str:
+        access_token_obj = self._http_request(
+            method='POST',
+            full_url=AUTHENTICATION_FULL_URL,
+            headers={'Content-Type': 'application/x-www-form-urlencoded'},
+            data=json_data
+        )
+
+        new_access_token = access_token_obj.get('access_token', '')
+        expire_in = arg_to_number(access_token_obj.get('expires_in')) or 1
+        self.store_token_in_context(new_access_token, expire_in)
+
+        demisto.log(f"Created new token: {new_access_token=}, {expire_in=}")
+        return new_access_token
+
+    def store_token_in_context(self, token: str, expire_in: int) -> None:
+        expire_date = get_current_time() + timedelta(seconds=expire_in) - timedelta(minutes=MINUTES_BEFORE_TOKEN_EXPIRED)
+        set_integration_context(context={
+            'token': token,
+            'expires': str(expire_date)
+        })
+
+    def get_events(self, start_date: str = "", end_date: str = "", continuation_token: str = "") -> dict:
```

Comment:
how does the API behave if we pass empty string to the API params below?

Reply:
If no parameters are provided, it will return the last 30 days of events.
but, since i have default start/end date this situation could not happen